packages/fxa-auth-server/test/local/senders/email.js

```diff
@@ -16,216 +16,790 @@ const sinon = require('sinon');
+      { test: 'include', expected: `href="${config.smtp.privacyUrl}?utm_medium=email&utm_campaign=fx-new-signin&utm_content=fx-privacy"` },
+      { test: 'include', expected: `href="${config.smtp.verifyLoginUrl.replace('//', `//${config.smtp.prependVerificationSubdomain.subdomain}.`)}?code=${MESSAGE.code}&uid=${MESSAGE.uid}&service=${MESSAGE.service}&utm_medium=email&utm_campaign=fx-new-signin&utm_content=fx-confirm-signin"` },
+      { test: 'include', expected: `IP address: ${MESSAGE.ip}` },
+      { test: 'include', expected: `${MESSAGE.location.city}, ${MESSAGE.location.stateCode}, ${MESSAGE.location.country} (estimated)` },
+      { test: 'include', expected: `${MESSAGE.uaBrowser} on ${MESSAGE.uaOS} ${MESSAGE.uaOSVersion}` },
+      { test: 'notInclude', expected: 'utm_source=email' },
+    ]],
+    ['text', [
+      { test: 'include', expected: `please change your password. ${config.smtp.initiatePasswordChangeUrl}?email=${encodeURIComponent(MESSAGE.email)}&utm_medium=email&utm_campaign=fx-new-signin&utm_content=fx-change-password` },
+      { test: 'include', expected: `Mozilla Privacy Policy ${config.smtp.privacyUrl}?utm_medium=email&utm_campaign=fx-new-signin&utm_content=fx-privacy` },
+      { test: 'include', expected: `${config.smtp.verifyLoginUrl.replace('//', `//${config.smtp.prependVerificationSubdomain.subdomain}.`)}?code=${MESSAGE.code}&uid=${MESSAGE.uid}&service=${MESSAGE.service}&utm_medium=email&utm_campaign=fx-new-signin&utm_content=fx-confirm-signin` },
+      { test: 'include', expected: `IP address: ${MESSAGE.ip}` },
+      { test: 'include', expected: `${MESSAGE.location.city}, ${MESSAGE.location.stateCode}, ${MESSAGE.location.country} (estimated)` },
+      { test: 'include', expected: `${MESSAGE.uaBrowser} on ${MESSAGE.uaOS} ${MESSAGE.uaOSVersion}` },
+      { test: 'notInclude', expected: 'utm_source=email' },
+    ]],
+  ])],
+  ['verifyPrimaryEmail', new Map([
+    ['subject', { test: 'equal', expected: 'Verify Primary Email' }],
+    ['headers', new Map([
+      ['X-Link', { test: 'equal', expected: `${config.smtp.verifyPrimaryEmailUrl}?code=${MESSAGE.code}&uid=${MESSAGE.uid}&type=primary&primary_email_verified=${encodeURIComponent(MESSAGE.email)}&service=${MESSAGE.service}&utm_medium=email&utm_campaign=fx-welcome-primary&utm_content=fx-activate` }],
+      ['X-SES-MESSAGE-TAGS', { test: 'equal', expected: sesMessageTagsHeaderValue('verifyPrimaryEmail') }],
+      ['X-Template-Name', { test: 'equal', expected: 'verifyPrimaryEmail' }],
+      ['X-Template-Version', { test: 'equal', expected: TEMPLATE_VERSIONS.verifyPrimaryEmail }],
+      ['X-Verify-Code', { test: 'equal', expected: MESSAGE.code }],
+    ])],
+    ['html', [
+      { test: 'include', expected: `href="${config.smtp.initiatePasswordChangeUrl}?email=${encodeURIComponent(MESSAGE.email)}&utm_medium=email&utm_campaign=fx-welcome-pri
```

Comment:
And there's a spot for trailhead-specific assertions too, although presumably these can all be unified once that special-casing is taken out.